src/lib/config-db.ts

```diff
@@ -215,3 +220,34 @@ export async function areSubdomainsSupported (logger?: ComponentLogger): Promise
   }
   return false
 }
+
+export async function isConfigSet (logger?: ComponentLogger): Promise<boolean> {
+  const log = logger?.forComponent('is-config-set')
+  try {
+    await configDb.open()
+    const config = await configDb.getAll()
+    // ignore private/app-only fields
+    return Object.keys(config).filter(key => !['_supportsSubdomains'].includes(key)).length > 0
+  } catch (err) {
+    log?.('error loading config from db', err)
+  } finally {
+    configDb.close()
+  }
+  return false
+}
+
+export async function compressConfig (config: ConfigDb): Promise<string> {
+  const timestamp = Date.now()
+  const configJson = JSON.stringify({ config, timestamp })
+
+  return LZString.compressToEncodedURIComponent(configJson)
+}
+
+export async function decompressConfig (compressedConfig: string): Promise<ConfigDb> {
+  const { config, timestamp } = JSON.parse(LZString.decompressFromEncodedURIComponent(compressedConfig))
+  // if the config is more than 10 seconds old, throw an error
+  if (timestamp < Date.now() - 1000) {
```

Comment:
1 second delay makes it very challenging to inject configs into users' subdomains, but I'm sure we could be a little more robust here.

cc @lidel